### ISSUE TITLE
Multi Piston Rework

### ISSUE THREAD

**Issue by Moobien** [ldtteam/Structurize#421]
### Prerequisites 

- [x] I am running the latest **alpha** version of Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my issue is not covered there.
- [x] I made sure that **this change request is not a duplicate of any existing change request.**

### Context 

- Minecraft Version: 1.1615
- Structurize Version: 208

## Enhancement or Change description

I have found that the multi piston is a great and useful block but how it currently functions is annoying and prone to issues.  Most 'sided' blocks in other mods allow inuitive sitting on which side things happen. The multi piston does not do this and its buttons rely on cardinal directions, the problem with this is if you rotate a structure with the multi piston in it, the cardinal direction it was set to is no longer valid and as such breaks the build.  I have also noticed issues where I have to fiddle a LOT to try to get it to function correctly (refer to included schematic) but then when its pasted into a new world it completely messes up and instead of doing what I want it to do on redstone on, it does the opposite or vice versa.

What I would like to see at minimum for it is a marked 'front' side and the cardinal directions changed to side directions. this will make using the multi piston more managable and intuitive to work with. Another thing I would like to see is a second set of buttons in the gui specifically for the direction that the multi piston 'retracts' the blocks to, this could be moot if the multi piston had an internal storage for the blocks it retracts, this could prove benefical as it would reduce the overall size of space needed to work with the multi piston.

### Notes or related things

[vault200.zip](https://github.com/ldtteam/Structurize/files/6997248/vault200.zip)

in the schematic, travel through the tunnel in the light placeholder blocks to get into the structure, on the left side is a dropper with a button in it. that button controls the multi pistons controlling the ramp and its side walls. in the windowed room above is a lever which controls a second set of multi pistons slide quartz blocks in and out of an alcove to block off the stairs leading deeper.

when the button is pressed, the yellow concrete blocks and polished andesite slabs are SUPPOSED to retract to the left (relative to the button when looking at the ramp)  until 1 row of them is still exposed. at the same time a second set retracts the diorite walls to the right (relative) leaving 1 still in front of the multi piston.

the quartz blocks that the multi pistons moveare supposed to move from 1 block in 'front' of the multi piston all the way to be flush with the top of the stairs.

---
## Viewers

* Add a thumbs-up to the change request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Shadizar** [comment]
I have a schematic that use them; works fine in my schematic world and doesn't work when built by builder
minecolonies-1.19.2-1.0.1207-ALPHA
multipiston-1.19.2-1.2.21-ALPHA
structurize-1.19.2-1.0.472-BETA
![multipistonbug](https://user-images.githubusercontent.com/94550408/215227178-9b08a8fb-30e3-4e31-8aed-0f914730738a.png)

**Comment by Shadizar** [comment]
I did not mirror the placement but likely did rotate it

**Comment by Moobien** [comment]
It should rotate fine, has rotated fine for me since the rework

**Comment by Shadizar** [comment]
Not sure what is wrong then. Clearly something is broken as both me and Spumantii are hitting errors.
And I never, ever touch the Mirror control, having heard too many horror stories heh

**Comment by Shadizar** [comment]
Tested today with the indicated versions and with the latest Archwood Collective mechanic level one to level 2 upgrade (level 2 does the initial placement of the multipistons)
And, the pistons are misplaced and do not have the same colors facing the same (relative) ways, and the mechanism therefore is broken until fixed manually by the player

**Comment by Raycoms** [comment]
The colors outside never rotate, the ones inside are supposed to

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by Shadizar** [comment]
Not sure I follow?
See my image, the schematic has one color and built has another color,  and the config is not updated to compensate...so, when built, the system is broken 
I am 100% sure no usage of Mirror, I avoid that control completely

**Comment by Raycoms** [comment]
the built has another color is correct, what is wrong is that the UI should reflect this

**Comment by Shadizar** [comment]
Would that ui fix also cause the mechanism to function? 
Because in my built one it didn't function until I changed the configuration

**Comment by Raycoms** [comment]
yes

**Comment by Shadizar** [comment]
Thanks

**Comment by Raycoms** [comment]
fixed, needs multipiston upgrade

**Comment by Shadizar** [comment]
I have updated to the build with the change, and it is looking pretty good.
I tried three tests:
1/two clean builds with 90 degree rotation from one another. Both worked fine on initial installation and on upgrade to a second level, so that's cool, thanks!
2/ On one of the clean-build upgrades I left the mechanism in an 'open' state (secret door). The builder did what I hoped (the end state was a closed secret door, so the DO blocks that had been moved by the multipiston were repositioned to the schematic location, and the mechanism worked fine), so that's great too
3/ On the original build that I had performed prior to the fix, I closed the door, and I manually put one multipiston misconfigured. On a repair, the builder did not reset the multipiston so I had to manually fix it up. Probably a minor thing. I.e. I think repairs and upgrades may not check side config; maybe even intentionally? In any case even if it is a flaw it's not a big deal

So from my testing, my issue seems fixed, figured you would like to know. (Not sure if other users may want to test their use cases).
